influxdb/src/main/scala/akka/stream/alpakka/influxdb/scaladsl/InfluxDBFlow.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.influxdb.scaladsl
+
+import akka.NotUsed
+import akka.stream.alpakka.influxdb.{impl, InfluxDBSettings, InfluxDBWriteMessage, InfluxDBWriteResult}
+import akka.stream.scaladsl.Flow
+import org.influxdb.InfluxDB
+
+import scala.collection.immutable
+
+/**
+ * Scala API to create InfluxDB flows.
+ */
+object InfluxDBFlow {
+
+  def create[T](settings: InfluxDBSettings)(
+      implicit influxDB: InfluxDB
+  ): Flow[InfluxDBWriteMessage[T, NotUsed], InfluxDBWriteResult[T, NotUsed], NotUsed] =
+    Flow[InfluxDBWriteMessage[T, NotUsed]]
+      .batch(settings.batchSize, immutable.Seq(_))(_ :+ _)
+      .via(new impl.InfluxDBFlowStage[T, NotUsed](None, influxDB))
+      .mapConcat(identity)
+
+  def typed[T](clazz: Class[T], settings: InfluxDBSettings)(
+      implicit influxDB: InfluxDB
+  ): Flow[InfluxDBWriteMessage[T, NotUsed], InfluxDBWriteResult[T, NotUsed], NotUsed] =
+    Flow[InfluxDBWriteMessage[T, NotUsed]]
+      .batch(settings.batchSize, immutable.Seq(_))(_ :+ _)
+      .via(new impl.InfluxDBFlowStage[T, NotUsed](Some(clazz), influxDB))
+      .mapConcat(identity)
+
+  def createWithPassThrough[T, C](settings: InfluxDBSettings)(
+      implicit influxDB: InfluxDB
+  ): Flow[InfluxDBWriteMessage[T, C], InfluxDBWriteResult[T, C], NotUsed] =
+    Flow[InfluxDBWriteMessage[T, C]]
+      .batch(settings.batchSize, immutable.Seq(_))(_ :+ _)
+      .via(new impl.InfluxDBFlowStage[T, C](None, influxDB))
+      .mapConcat(identity)
+
+  def typedWithPassThrough[T, C](clazz: Class[T], settings: InfluxDBSettings)(
+      implicit influxDB: InfluxDB
+  ): Flow[InfluxDBWriteMessage[T, C], InfluxDBWriteResult[T, C], NotUsed] =
+    Flow[InfluxDBWriteMessage[T, C]]
+      .batch(settings.batchSize, immutable.Seq(_))(_ :+ _)
+      .via(new impl.InfluxDBFlowStage[T, C](Some(clazz), influxDB))
+      .mapConcat(identity)
+
+  def createWithContext[T, C](settings: InfluxDBSettings)(
```

Comment:
Great that you prepared for `withContext`, we'll see that more and more.
I wonder if we should default to using tuples instead of the pass-through field in the messages.

Reply:
Good! I suppose something like this

```scala


def typedWithContext[T, C](clazz: Class[T], settings: InfluxDBSettings)(
    implicit influxDB: InfluxDB
): Flow[((InfluxDBWriteMessage[T],NotUsed), C), (InfluxDBWriteResult[T], C), NotUsed] =
  Flow[((InfluxDBWriteMessage[T],NotUsed), C)]
    .map {
      case (wm, pt) =>
        InfluxDBWriteMessage(wm.point, pt)
    }
    .via(typedWithPassThrough(clazz, settings))
    .map { wr =>
      (wr, wr.writeMessage.passThrough)
    }

```

Is there any implementation that does so?

Reply:
We've added initial support in Alpakka Elasticsearch and have a usage example [in the documentation](https://doc.akka.io/docs/alpakka/current/examples/elasticsearch-samples.html#example-read-from-a-kafka-topic-and-publish-to-elasticsearch). It is implemented on top of the pass-through just as you did.
The upcoming Akka release 2.5.23 will fix a few things around the Java `withContext` API. We'll upgrade Alpakka immediately when that release is out.